### ISSUE TITLE
REST API `LedgerStateDelta.Txleases` contains invalid JSON

### ISSUE THREAD

**Issue by jasonpaulos** [algorand/go-algorand#6097]
## Background
The `ledgercore.StateDelta.Txleases` field is defined as follows:

https://github.com/algorand/go-algorand/blob/c6a433b5035377246340710a91b295553bb6ac1f/ledger/ledgercore/statedelta.go#L103-L105

Where `Txlease` is a struct defined as:

https://github.com/algorand/go-algorand/blob/c6a433b5035377246340710a91b295553bb6ac1f/ledger/ledgercore/statedelta.go#L64-L69

## Problem
Using an object as a map key is a defined behavior for msgpack (although it does have its own complications in other languages), but JSON prohibits it. If you attempt to ask the REST API for a JSON-encoded ledger state delta, and this field is populated, the response will contain invalid JSON.

For example, this state delta,

```go
ledgercore.StateDelta{
	Txids: map[transactions.Txid]ledgercore.IncludedTransactions{
		{1, 2, 3}: {LastValid: 7, Intra: 1},
	},
	Txleases: map[ledgercore.Txlease]basics.Round{
		{Sender: basics.Address{9, 9, 9}, Lease: [32]byte{12, 13, 14}}: 8,
	},
}
```

encodes to this invalid JSON:

```json
{
  "Accts": {
    "Accts": null,
    "AppResources": null,
    "AssetResources": null
  },
  "Creatables": null,
  "Hdr": null,
  "KvMods": null,
  "PrevTimestamp": 0,
  "StateProofNext": 0,
  "Totals": {},
  "Txids": {
    "AQIDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=": {
      "Intra": 1,
      "LastValid": 7
    }
  },
  "Txleases": {
    {
  "Lease": "DA0OAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=",
  "Sender": "BEEQSAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADSLODMM"
}: 8
  }
}
```

## Solution

A possible solution is to create a variant of the `ledgercore.StateDelta` struct designed to be JSON serialized, perhaps by transforming the `Txleases` map into an array.

**Comment by Ashy5000** [comment]
What about creating new `MarshalJSON()` and `UnmarshalJSON()` methods on `ledgercore.StateDelta`? `MarshalJSON()` could convert the regular `ledgercore.StateDelta` value to a JSON-serializable variant, then serializing the converted value to a string. `UnmarshalJSON()` would do the opposite.

This way conversion to and from the variant of `ledgercore.StateDelta` is completely contained within the standard `json.Marshal()` and `json.Unmarshal()` functions.

**Comment by Ashy5000** [comment]
I've implemented this in PR #6100.
